File: LeetCode/Array/code/CPP/reshape-the-matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        int n = nums.size(), m = nums[0].size();
        if(n * m < r * c)
            return nums;
        vector<vector<int>> ans(r, vector<int>(c));
        int ni = 0, mj = 0;
        for(int i=0;i<r;i++)
        {
            for(int j=0;j<c;j++)
            {
                ans[i][j] = nums[ni][mj];
                mj++;
                if(mj == m)
                {
                    mj = 0;
                    ni++;
                }
            }
        }
        return ans;
    }
};

File: LeetCode/Array/code/Python/reshape-the-matrix.py
class Solution:
    def matrixReshape(self, nums, r, c):
        """
        :type nums: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """
        n, m = len(nums), len(nums[0])
        if r * c > n * m:
            return nums
        ans = list()
        ni, mj = 0, 0
        for i in range(r):
            ans.append([])
            for j in range(c):
                ans[i].append(nums[ni][mj])
                mj += 1
                if mj == m:
                    ni += 1
                    mj = 0
        return ans